roles/openshift_certificate_expiry/library/openshift_cert_expiry.py

```diff
@@ -129,7 +131,7 @@ def _parse_cert(self):
                 not_after_raw = l.partition(' : ')[-1]
                 # Last item: ('Not After', ' : ', 'Feb  7 18:19:35 2019 GMT')
                 not_after_parsed = datetime.datetime.strptime(not_after_raw, '%b %d %H:%M:%S %Y %Z')
-                self.not_after = not_after_parsed.strftime('%Y%m%d%H%M%SZ')
+                self.not_after = not_after_parsed.strftime('%Y%m%d%H%M%SZ').encode()
```

Comment:
I definitely think this needs more scrutiny on the encoding/decoding aspects.

The main reason that I ended up down this path is because of using pyOpenSSL for the tests, where parameters and return values are bytes, which caused comparison failures between the fake class and the actual class.

Since we are trying to mimic the behavior of pyOpenSSL we need to copy the interface they have, so I'm not sure handling encode/decode on serialization is the answer, since we'd still have the potential for different behavior in py3 depending on using openssl or the fake class.